packages/story-editor/src/components/library/panes/media/common/paginatedMediaGallery.js

```diff
@@ -80,6 +80,17 @@ function PaginatedMediaGallery({
       return;
     }
 
+    // Load the next page if the page isn't full, ie. scrollbar is not visible.
+    if (node.clientHeight === node.scrollHeight) {
+      setNextPage();
+      return;
+    }
+
+    // If scrollTop is zero, we know we do not need to fetch an additional page yet.
+    if (node.scrollTop === 0) {
+      return;
```

Comment:
So here is the issue with this test being flakey. When this is first ran `node` gets defined, but sometimes the media elements are still being fetched resulting in the `scrollTop`  being set at `0`. ie. The container that is the node _is_ there, but it doesn't have any contents yet. The early return on line 90 `if node.scrollTop === 0` would solve the issue, _but_ the early return above on line 84 prevents it. This is because when the `node.ScrollTop` is `0` the `node.clientHieght === node.scrollHieght` is true. Therefore we call `setNextPage` in order to fill the container space with media elements. This means we make two fetch media calls when really we only want to make one. I think in the before times when the api was fetching 20 items, it was possible that 20 item may not have entirely filled the container's height. Which would totally make sense that we would want to make an additional call. We always want to fill that space. However, now that the api returns 50 items, it is very unlikely the container's height would not be full once the first batch of items has been called. In fact, I had to turn my external monitor vertical and  set resolution to super tiny in order to see that the second call would be needed. And the site as a whole is unusable with it that small. So I propose that we remove the check which makes the second call on line 90 altogether. Or I somehow wait for the elements to actually be present and from there check to see if we still need a second to fill the remaining space. @BrittanyIRL thoughts?

Reply:
thank you for such an insightful summary and for the time you've spent getting to the bottom of this. I'm fine removing that check, but I want to run this past Miina since her team has more experience and context for the media panels. @miina does the above make sense to ya?

Reply:
SGTM, however, pinging @swissspidy here as well since Infra pod that has done a lot of work with everything related to Media.

Reply:
Unfortunately I don't have more info here as this logic here was implemented by our temp media pod last year